src/genie/libs/parser/iosxe/tests/ShowBgpAllNeighbors/cli/equal/golden_output7_expected.py

```diff
@@ -0,0 +1,195 @@
+expected_output = {
+    "list_of_neighbors": ["10.10.11.1"],
+    "vrf": {
+        "CustA-VPN1": {
+            "neighbor": {
+                "10.10.11.1": {
+                    "address_family": {
+                        "ipv4 multicast": {},
+                        "l2vpn evpn": {},
+                        "mvpnv4 unicast": {},
+                        "mvpnv6 unicast": {},
+                        "vpnv4 flowspec": {},
+                        "vpnv4 multicast": {},
```

Comment:
Why are these dicts empty?

Reply:
Hi @Taarini, the dicts are empty as the show output for these BGP Adress families is empty as the bgp config does not have these specific AF enabled.
You can see the same in several other showBgp related golden outputs e.g. golden_output1_expected.py line 531 - 537

Note I am not the original author of this parser - I think it should not be changed for backward compatibility and I think having a empty dict for a not enabled BGP AF is correct.

Reply:
I understand that you may be the original author but see if you can add a condition to update these values only if the keys are present else don't update the dict. Doing this will not affect the backward compatibility as you are avoiding the keys whose values are empty.

Reply:
from my point of view it is even a good thing to have this keys with empty dict, as displaying this keys/(empty) dicts still adds value to the parser, as this indicates from the original bgp output, the router itself is capable to negotiate the corresponding BGP Address families, but neighbor config for these AF is not setup for this neighbor. Omitting the empty keys completly will remove some value info from the parsed output.
Of do you mean adding the keys with an "None" value? Would this make more sense ?

Reply:
No that is not what i mean. I meant don't create empty dict unless it has a values. 
Below is what I am suggesting,

```suggestion
            "neighbor": {
                "10.10.11.1": {
                    "address_family": {
                        "vpnv4 unicast": {
                            "current_time": "0x1D0B600",
                            "last_read": "00:00:20",
                            "last_write": "00:00:16",
                            "session_state": "Established",
                            "up_time": "00:19:52",
                        }
                    },
```
From the above it is evident that rest values are not present.
But below is what you are showing as part of parsed dict,
```suggestion
            "neighbor": {
                "10.10.11.1": {
                    "address_family": {
                        "ipv4 multicast": {},
                        "l2vpn evpn": {},
                        "mvpnv4 unicast": {},
                        "mvpnv6 unicast": {},
                        "vpnv4 flowspec": {},
                        "vpnv4 multicast": {},
                        "vpnv4 unicast": {
                            "current_time": "0x1D0B600",
                            "last_read": "00:00:20",
                            "last_write": "00:00:16",
                            "session_state": "Established",
                            "up_time": "00:19:52",
                        }
                    },
```

Reply:
Understood you intend, I can do the change (maybe separate PR?) but I still think this will remove valuable info from the parsed output, also the dict is empty the presence of the key itself indicates already bgp capability of the router itself - like in the original "show bgp all neighbor" output:

some output from bgp speaker IOSXE16.9.
```
(...)
For address family: VPNv4 Unicast

For address family: IPv4 Multicast

For address family: L2VPN E-VPN

For address family: VPNv4 Multicast

For address family: MVPNv4 Unicast

For address family: VPNv4 Flowspec

(...)
```

and for comparison some output from bgp speaker running  IOSXE17.2

```
(...)
For address family: VPNv4 Unicast

For address family: IPv4 Multicast

For address family: L2VPN E-VPN

For address family: VPNv4 Multicast

For address family: MVPNv4 Unicast

For address family: MVPNv6 Unicast

For address family: VPNv4 Flowspec
(...)
```
=> the router with 17.2 has an additional AF MVPNv6 which is missing from the other router - also the output is empty  as no config is found, the output indicates the bgp AF capability of the router.
If we remove the empty keys alltogether we will loose this info from the parsed output.
having the key with empty dict present indicates the local router has the specific BGP AF capability but it is currently not configured.